### PR TITLE
feature: Better error messages for ad_auth

### PR DESCRIPTION
**Pull request by murrant**
#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

### CODE REVIEW

File: html/includes/authentication/active_directory.inc.php [hunk 1]
```diff
@@ -4,10 +4,14 @@
 
 // disable certificate checking before connect if required
 if (isset($config['auth_ad_check_certificates']) &&
-          $config['auth_ad_check_certificates'] == 0) {
+          !$config['auth_ad_check_certificates']) {
```
Comment (laf) [inline html/includes/authentication/active_directory.inc.php]: Not better to check for !== true?

File: html/includes/authentication/active_directory.inc.php [hunk 2]
```diff
@@ -4,10 +4,14 @@
...
     putenv('LDAPTLS_REQCERT=never');
 };
 
+if (isset($config['auth_ad_debug']) && $config['auth_ad_debug']) {
```
Comment (laf) [inline html/includes/authentication/active_directory.inc.php]: === true?

File: html/includes/authentication/active_directory.inc.php [hunk 3]
```diff
@@ -17,14 +21,14 @@
...
-    global $config, $ldap_connection;
+    global $config, $ldap_connection, $auth_error;
 
     if ($ldap_connection) {
         // bind with sAMAccountName instead of full LDAP DN
         if ($username && ldap_bind($ldap_connection, "{$username}@{$config['auth_ad_domain']}", $password)) {
             // group membership in one of the configured groups is required
             if (isset($config['auth_ad_require_groupmembership']) &&
-                $config['auth_ad_require_groupmembership'] > 0) {
+                $config['auth_ad_require_groupmembership']) {
```
Comment (laf) [inline html/includes/authentication/active_directory.inc.php]: === true?

**Comment by laf** [comment]
I put a couple of comments in but in general is it not better to do explicit checks like `=== true` or `!== true` rather than isset or just an if check?

**Comment by laf** [comment]
If not btw then this is fine for a merge so feel free to do so.

File: html/includes/authentication/active_directory.inc.php (hunk 1 above)
Comment (murrant) [inline html/includes/authentication/active_directory.inc.php]: Users have `$config['auth_ad_check_certificates'] = 1` in their configs right now.  I wanted to support true and false too.

**Comment by murrant** [comment]
We need to allow for 1 and 0 settings too, because that is what the docs had before.

**Pull request merged**